selftests/unit/test_task_runtime.py

```diff
@@ -1,7 +1,42 @@
+import os
 from unittest import TestCase
 
 from avocado.core.nrunner import Runnable, Task
-from avocado.core.task.runtime import RuntimeTask
+from avocado.core.suite import TestSuite
+from avocado.core.task.runtime import RuntimeTask, RuntimeTaskGraph
+from avocado.utils import script
+from selftests.utils import TestCaseTmpDir
+
+SINGLE_REQUIREMENT = '''#!/usr/bin/env python3
+from avocado import Test
```

Comment:
Thanks for finding this, I will investigate it.

Reply:
There was a problem with `TestID` do, the runner wasn't able to identify requirements. I did a quick fix by setting the identifier prefix for requirements to 0, but we need to do changes to `TestID`.

Reply:
I can confirm the fix worked, I just ran the selftests and it didn't get stuck.

Reply:
@ana actually the problem wasn't with selftests but with the 'SINGLE_REQUIREMENT` example itself. If you create test form 'SINGLE_REQUIREMENT` and run avocado like `avocado run single_requirment.py` then you will get stuck on master, but on this branch it will be ok. Also be aware that this test will install `hello` package to your system. 

We don't have such test in selftests because of #4620